### ISSUE TITLE
[BUG] - autotunnel not working, location perms seemingly not accessed in several days

### ISSUE THREAD

**Issue by mdecourcy** [zaneschepke/wgtunnel#472]
**Describe the bug**
WGTunnel not auto connecting after disconnecting from WiFi. It appears it hasn't accessed location data since 11/30 

**Smartphone (please complete the following information):**

- Device: pixel 9
- Android Version: Android 15 graphene os build 2024112700  
- App Version: 3.6.1
- Backend: [e.g. Kernel, Userspace] 

**To Reproduce**

1. Have autotunnel on for untrusted wifis of mobile data
2. Disconnect from WiFi (either via settins or walking out of range of router)
3. You never connect to the vpn

**Expected behavior**
Autotunnel gets triggered after leaving/joining network

**Screenshots (Only if necessary)**
![Screenshot_20241202-175232.png](https://github.com/user-attachments/assets/70c5382f-1dbf-4f6d-ae22-52e0afb40d7c)

**Additional context**
Error logs (unsure if related)

```
2024-12-03T01:57:01.000000655Z 24108 26309 ERROR AmneziaWG/home-proxmox: message= UAPIOpen: mkdir /data/data/org.amnezia.awg: permission denied
2024-12-03T01:57:01.000000688Z 24108 26288 ERROR cutils-trace: message= Error opening trace file: No such file or directory (2)
```

Note it was previously working. Unsure if an app update or android update broke something.

**Comment by mdecourcy** [comment]
Appears rebooting the device fixed the auto tunneling on WiFi disconnect

**Comment by timi35** [comment]
Same here in a Samsung s23 Ultra.
It seems the reason is the last Samsung Update. A restart didn't help.

**Comment by timi35** [comment]
I tested it again. In the last nightly AutoTunnel doesn't work.
In Version fecc55f it works better, but the tunnel doesn't stop if there is no connection.

**Comment by zaneschepke** [comment]
Hey! The latest nightly coming out tonight should fix a lot of these issues. Please let me know if it helps. Thanks!

**Comment by timi35** [comment]
Hi Zane,

 

this version has the same problems on my device.

 

1.	It still connects to whitelisted WLANs. 
2.	It doesn´t disable VPN if there is no connection.  It looks like keep alive, but I don’t use this feature.

 

Btw: if I add a wlan to the whitelist, VPN disconnect immediately.

But as soon as I reconnect to the whitelisted WLAN, VPN is active.

 

Comment to 1.:

I tested it again, the problem is reproducible:

 

First connection to a whitelisted WLAN – VPN starts

Second connection to a whitelisted WLAN – VPN doesn’t start -> correct

Third connection to a whitelisted WLAN – VPN starts

and so on…

 

Best regards!

 

Von: Zane Schepke ***@***.***> 
Gesendet: Sonntag, 8. Dezember 2024 02:30
An: zaneschepke/wgtunnel ***@***.***>
Cc: timi35 ***@***.***>; Comment ***@***.***>
Betreff: Re: [zaneschepke/wgtunnel] [BUG] - autotunnel not working, location perms seemingly not accessed in several days (Issue #472)

 

Hey! The latest nightly coming out tonight should fix a lot of these issues. Please let me know if it helps. Thanks!

**Comment by zaneschepke** [comment]
Thanks for this detail, I'm able to reproduce. Working on a fix now!

**Comment by zaneschepke** [comment]
I found the bug. This should be fixed in the next nightly.

**Comment by timi35** [comment]
Dear Zane,

 

it looks great until now. I’ll test it still a while.

Btw, now I can see the „missing“ Feature in Autotunneln 😊 and know why WGtunnel didn’t stop the tunnel 😉

 

Thanks a lot!

 

Best regards,

Tim

 

Von: Zane Schepke ***@***.***> 
Gesendet: Sonntag, 8. Dezember 2024 21:08
An: zaneschepke/wgtunnel ***@***.***>
Cc: timi35 ***@***.***>; Comment ***@***.***>
Betreff: Re: [zaneschepke/wgtunnel] [BUG] - autotunnel not working, location perms seemingly not accessed in several days (Issue #472)

 

I found the bug. This should be fixed in the next nightly.

**Comment by zaneschepke** [comment]
I think this one is now resolved by 3.6.3. Marking this as closed. We can reopen if the issue comes back up.

**Comment by mdecourcy** [comment]
@zaneschepke noticing this issue again
